### PR TITLE
Dynamic Loading Improvements

### PR DESCRIPTION
**Pull request by dinomut1**
*duplicate object now works on object3Ds
*add SceneObjectInput component
*dynamic scene loading now correctly handles children of dynamic nodes
*add lock toggle to properties panel

### CODE REVIEW

File: packages/engine/src/scene/systems/SceneObjectDynamicLoadSystem.ts [hunk 1]
```diff
@@ -63,7 +82,34 @@ export default async function SceneObjectDynamicLoadSystem(world: World) {
             position: data.position
           })
 
+          hasComponent(entity, RigidBodyComponent) && Physics.removeCollidersFromRigidBody(entity, world.physicsWorld)
```
Comment (HexaField) [inline packages/engine/src/scene/systems/SceneObjectDynamicLoadSystem.ts:85]: this should be happening in the physics system, not here

File: packages/engine/src/scene/systems/SceneObjectDynamicLoadSystem.ts [hunk 2]
```diff
@@ -63,7 +82,34 @@ export default async function SceneObjectDynamicLoadSystem(world: World) {
...
+          const targetNode = nodeMap.get(entity)
+          if (targetNode) {
+            iterateEntityNode(targetNode, (node) => {
```
Comment (HexaField) [inline packages/engine/src/scene/systems/SceneObjectDynamicLoadSystem.ts:88]: what is the case in which a child entity is separated in distance to it's parent? the dynamic load should only ever be used in leaf nodes of the entity tree, or alternatively parent nodes should with dynamic loading should stop their children from being loaded, and then handle the loading/unloading of their children

File: packages/engine/src/scene/systems/SceneObjectDynamicLoadSystem.ts [hunk 3]
```diff
@@ -63,7 +82,34 @@ export default async function SceneObjectDynamicLoadSystem(world: World) {
...
+                    const data = { name: node.uuid, components: serializeEntity(node.entity) }
+                    if (node.parentEntity) {
+                      const parentNode = nodeMap.get(node.parentEntity!)!
+                      data['parent'] = parentNode.uuid
+                    }
+                    createSceneEntity(node.uuid, data)
+                  }
+                )
+
+              hasComponent(node.entity, RigidBodyComponent) &&
```
Comment (HexaField) [inline packages/engine/src/scene/systems/SceneObjectDynamicLoadSystem.ts:102]: once again, should be done in physics system

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/scene/systems/SceneObjectDynamicLoadSystem.ts (hunk 2 above)
Comment (dinomut1) [reply packages/engine/src/scene/systems/SceneObjectDynamicLoadSystem.ts:88]: Any child that has a position offset will be separated in distance from its parent. Used when wanting to dynamically load an object or group of objects as the player approaches a position in the level.

File: packages/engine/src/scene/systems/SceneObjectDynamicLoadSystem.ts (hunk 3 above)
Comment (dinomut1) [reply packages/engine/src/scene/systems/SceneObjectDynamicLoadSystem.ts]: now in physics system

**Pull request merged**